File: Examples/ProductRecommendations/Obj-C/RecommendationsExample/DemoCell.m
//
//  DemoCell.m
//  RecommendationsExample
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

#import "DemoCell.h"
#import <SDWebImage/UIImageView+WebCache.h>

@interface DemoCell ()

@property IBOutlet UILabel *productName;
@property IBOutlet UILabel *price;
@property IBOutlet UILabel *numReview;
@property IBOutlet UILabel *rating;
@property IBOutlet UIImageView *productImageView;
@property IBOutlet HCSStarRatingView *starRating;

@end

@implementation DemoCell

- (void)setBvRecommendedProduct:(BVRecommendedProduct *)bvRecommendedProduct {

  super.bvRecommendedProduct = bvRecommendedProduct;

  NSURL *url = [NSURL URLWithString:self.bvRecommendedProduct.imageURL];
  [self.productImageView sd_setImageWithURL:url];
  self.productName.text = bvRecommendedProduct.productName;
  self.rating.text = [NSString
      stringWithFormat:@"%.1f",
                       [bvRecommendedProduct.averageRating floatValue]];
  self.starRating.value = [bvRecommendedProduct.averageRating floatValue];
  self.numReview.text = [NSString
      stringWithFormat:@"(%ld)", [bvRecommendedProduct.numReviews longValue]];
  self.price.text = bvRecommendedProduct.price;
}

@end

File: Examples/ProductRecommendations/Swift/RecommendationsExample/DemoCell.swift
//
//  DemoCell.swift
//  RecommendationsExample
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

import BVSDK
import HCSStarRatingView
import SDWebImage

class DemoCell: BVRecommendationCollectionViewCell {
    
    @IBOutlet weak var productName : UILabel!
    @IBOutlet weak var price : UILabel!
    @IBOutlet weak var numReview : UILabel!
    @IBOutlet weak var rating : UILabel!
    @IBOutlet weak var productImageView : UIImageView!
    @IBOutlet weak var starRating : HCSStarRatingView!
    
    override var bvRecommendedProduct: BVRecommendedProduct! {
        
        didSet {
            
            print("Prod: " + bvRecommendedProduct.debugDescription)
            
            let imageUrl = NSURL(string: bvRecommendedProduct.imageURL)
            self.productName.text = bvRecommendedProduct.productName
            self.rating.text = "\(bvRecommendedProduct.averageRating.floatValue)"
            self.numReview.text = "(\(bvRecommendedProduct.numReviews.intValue) reviews)"
            self.starRating.value = (CGFloat)(bvRecommendedProduct.averageRating.floatValue)
            self.productImageView?.sd_setImage(with: imageUrl as URL!)
            
        }
        
    }
    
}
